### PR TITLE
docs(grid): Document breaking change in column template component usage

### PR DESCRIPTION
**Pull request by dimodi**

### CODE REVIEW

File: components/grid/templates/column.md [hunk 1]
```diff
@@ -86,7 +96,91 @@ The example below shows how to:
...
 
->tip The above example renders read-only checkboxes to display boolean values. It is possible to [use checkboxes in display mode and directly change the underlying data source values]({%slug grid-kb-checkbox-editing%}). This can make boolean value editing faster, because the Grid doesn't go into edit mode.
+>tip The above example renders read-only checkboxes to display boolean values. You can also [use checkboxes in display mode and directly change the underlying data source values]({%slug grid-kb-checkbox-editing%}). This can make boolean value editing faster, because the Grid doesn't go into edit mode.
+
+## Using Components in Grid Column Templates
+
+The Grid optimizes the UI renders after data operations. If you are using child components inside Grid column templates, <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/components/element-component-model-relationships" target="_blank">set the `@key` attribute to these components to ensure that they always show the correct values and content after filtering, paging, and sorting</a>.
+
+>caption Seting @key to child components inside a Grid column template
+
```
Comment (ntacheva) [inline components/grid/templates/column.md:106]: <div class="skip-repl"></div>

Comment (ntacheva) [inline components/grid/templates/column.md:106]:
```suggestion

<div class="skip-repl"></div>
```

**Review by ntacheva — APPROVED** [review]

File: components/grid/templates/column.md [hunk 2]
```diff
@@ -10,16 +10,26 @@ position: 5
...
+By default, each Grid cell renders the value of the respective column `Field` of the current data item (row). You can change this behavior by using a column `Template` and adding your own content or logic.
 
-The column template (the `<Template>` tag under the column definition) is what the Grid uses to show the "view" representation of the cell. This also includes a column that is marked as `Editable="false"` and is in edit mode.
+## Basics
+
+To define a column template, use a `<Template>` tag inside the `<GridColumn>` tag. The Grid uses the defined `Template` to show the "view" representation of all cells in that column. This also includes cells from [columns that are marked as `Editable="false"`]({%slug components/grid/columns/bound%}#data-operations), while the cells' parent row is in [inline edit mode]({%slug components/grid/editing/inline%}).
+
+Visual Studio tends to autocomplete the `<Template>` tag with a lowercase `t` which breaks the template logic and does not allow you to access the `context`. Ensure the `Template` tag uses a capital `T`.
```
Comment (ikoevska) [inline components/grid/templates/column.md:19]: This seems like a rather important note that needs more attention from the readers. I recommend applying an appropriate callout formatting style.

File: upgrade/breaking-changes/7-0-0.md [hunk 3]
```diff
@@ -44,6 +44,40 @@ The obsolete `ClearButton` parameter is removed. Use `ShowClearButton` instead.
 * The obsolete `AutoFitColumns()` method is removed. Use `AutoFitColumnsAsync()` instead.
 * The obsolete `AutoFitAllColumns()` method is removed. Use `AutoFitAllColumnsAsync()` instead.
 * When using [grouping and `OnRead`]({%slug components/grid/manual-operations%}#grouping-with-onread), casting `DataSourceResult.Data` to a list of objects (`.Cast<object>()`) is no longer needed.
+* [Components in a Grid column `<Template>` require a `@key`]({%slug grid-templates-column%}#using-components-in-grid-column-templates) in order to display correct values after data operations like sorting, filtering, paging, and others.
```
Comment (ikoevska) [inline upgrade/breaking-changes/7-0-0.md:47]: You can almost always use just "to" instead of "in order to".

**Review by ikoevska — COMMENTED** [review]
Reviewed.

**Pull request merged**